File: Java/594. Longest Harmonious Subsequence.java
class Solution {
    public int findLHS(int[] nums) {
        Map<Integer, Integer> count = new HashMap<>();
        int ans = 0;

        for (int n: nums) {
            count.put(n, count.getOrDefault(n, 0) + 1);
            if (count.containsKey(n - 1)) {
                ans = Math.max(ans, count.get(n) + count.get(n - 1));
            }
            if (count.containsKey(n + 1)) {
                ans = Math.max(ans, count.get(n) + count.get(n + 1));
            }
        }

        return ans;
    }
}

File: Python3/594. Longest Harmonious Subsequence.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        count = Counter(nums)
        ans = 0

        for key, val in count.items():
            if key - 1 in count:
                ans = max(ans, val + count[key - 1])
            
            if key + 1 in count:
                ans = max(ans, val + count[key + 1])

        return ans